citizennet/cn-ui
# API-2909 Truncated Text Expansion Toggle

## Jira Ticket

- [https://citizennet.atlassian.net/browse/API-2909](https://citizennet.atlassian.net/browse/API-2909)

## What does this pull request do?

<img width="413" alt="Screenshot 2023-03-14 at 12 34 19 PM" src="https://user-images.githubusercontent.com/75546183/225078498-21c8961e-f713-4474-af6f-78474fcdc8bc.png">

This PR adds a text expansion toggle to the `cn-truncate` directive. The button is functional and works in the same way as clicking the text that's being truncated. I had also kept changes minimal without having to update all `cn-truncate` elements.

Lastly, the latest changes Wildcat references on production exist in [this branch](https://github.com/citizennet/cn-ui/tree/API-2821-NEW-B), which was never merged to master and turned into a release. I've merged that branch into this one, and will create a new release after this PR is merged so our process is uniform again.

The only new changes part of this PR are in `cn-truncate.directive.js`.

## How should this be manually tested?

1. Navigate to the campaign targeting form or ad set details form.
2. Add a large amount of targeting, so the targeting sentence on the right-hand pane is truncated.
3. Verify a `Show more` button is displayed directly below the truncated text and is functional.
